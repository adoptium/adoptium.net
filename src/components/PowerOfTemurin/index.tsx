import React from "react"
import { Link } from "@/i18n/navigation"
import { useTranslations } from "next-intl"

import UiVirtualScroll from "@/components/UiVirtualScroll"

const PowerOfTemurin = () => {
  const t = useTranslations("PowerOfTemurin")
  return (
    <div className="bg-purple py-16">
      <div className="mx-auto max-w-3xl px-6 lg:px-8 flex flex-col items-center justify-center mb-16">
        <h2 className="text-center text-4xl lg:text-5xl leading-[44px] lg:leading-[56px] font-semibold text-white-900">
          {t("title")}
        </h2>
        <h3 className="text-center text-base leading-6 font-normal text-grey mt-6">
          {t("description")}
        </h3>
        <Link href="/business-benefits">
          <button className="bg-transparent mt-10 border-2 border-pink-500/0 text-white text-base leading-6 font-bold w-[191px] h-[48px] rounded-2xl gradient-border">
            {t("button")}
          </button>
        </Link>
      </div>
      <UiVirtualScroll />
    </div>
  )
}

export default PowerOfTemurin
