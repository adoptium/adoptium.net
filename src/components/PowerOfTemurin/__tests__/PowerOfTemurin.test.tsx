import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it, vi } from "vitest"
import PowerOfTemurin from ".."

vi.mock("../../../UiVirtualScroll", () => {
  return {
    default: () => <div>UiVirtualScroll</div>,
  }
})

describe("PowerOfTemurin component", () => {
  it("should render correctly", () => {
    const { container } = render(<PowerOfTemurin />)

    expect(container).toMatchSnapshot()
  })
})
