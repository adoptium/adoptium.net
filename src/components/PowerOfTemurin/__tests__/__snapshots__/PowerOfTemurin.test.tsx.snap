// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PowerOfTemurin component > should render correctly 1`] = `
<div>
  <div
    class="bg-purple py-16"
  >
    <div
      class="mx-auto max-w-3xl px-6 lg:px-8 flex flex-col items-center justify-center mb-16"
    >
      <h2
        class="text-center text-4xl lg:text-5xl leading-[44px] lg:leading-[56px] font-semibold text-white-900"
      >
        Text
      </h2>
      <h3
        class="text-center text-base leading-6 font-normal text-grey mt-6"
      >
        Text
      </h3>
      <a
        href="/business-benefits"
      >
        <button
          class="gradient-border bg-transparent mt-10 text-white text-base leading-6 font-bold min-w-[191px] px-6 h-[48px] rounded-2xl"
        >
          Text
        </button>
      </a>
    </div>
    <div
      class="lg:hidden max-w-[500px] mx-auto flex flex-col text-center items-center"
    >
      <img
        alt="Description"
        class="w-full mb-0 max-w-[366px] object-cover"
        data-nimg="1"
        decoding="async"
        height="366"
        loading="lazy"
        src="/images/icons/feature-layers.svg"
        style="color: transparent;"
        width="366"
      />
      <img
        alt="scroll divider"
        class="mb-0"
        data-nimg="1"
        decoding="async"
        height="80"
        loading="lazy"
        src="/images/icons/scroll-divider.svg"
        style="color: transparent;"
        width="16"
      />
      <div
        class="flex flex-col flex-1 px-4"
      >
        <h1
          class="text-xl my-5 font-bold"
        >
          Text
        </h1>
        <div
          class="text-base text-grey"
        >
          Text
        </div>
      </div>
    </div>
    <div
      class="lg:hidden max-w-[500px] mx-auto flex flex-col text-center items-center"
    >
      <img
        alt="Description"
        class="w-full mb-0 max-w-[366px] object-cover"
        data-nimg="1"
        decoding="async"
        height="366"
        loading="lazy"
        src="/images/icons/feature-layers.svg"
        style="color: transparent;"
        width="366"
      />
      <img
        alt="scroll divider"
        class="mb-0"
        data-nimg="1"
        decoding="async"
        height="80"
        loading="lazy"
        src="/images/icons/scroll-divider.svg"
        style="color: transparent;"
        width="16"
      />
      <div
        class="flex flex-col flex-1 px-4"
      >
        <h1
          class="text-xl my-5 font-bold"
        >
          Text
        </h1>
        <div
          class="text-base text-grey"
        >
          Text
        </div>
      </div>
    </div>
    <div
      class="lg:hidden max-w-[500px] mx-auto flex flex-col text-center items-center"
    >
      <img
        alt="Description"
        class="w-full mb-0 max-w-[366px] object-cover"
        data-nimg="1"
        decoding="async"
        height="366"
        loading="lazy"
        src="/images/icons/feature-layers.svg"
        style="color: transparent;"
        width="366"
      />
      <img
        alt="scroll divider"
        class="mb-0"
        data-nimg="1"
        decoding="async"
        height="80"
        loading="lazy"
        src="/images/icons/scroll-divider.svg"
        style="color: transparent;"
        width="16"
      />
      <div
        class="flex flex-col flex-1 px-4"
      >
        <h1
          class="text-xl my-5 font-bold"
        >
          Text
        </h1>
        <div
          class="text-base text-grey"
        >
          Text
        </div>
      </div>
    </div>
    <div
      class="lg:block hidden open-animation-wrapper"
    >
      <div
        class="max-w-[1048px] ui-virtual-wrapper mt-10 relative mx-auto px-4 sm:px-0"
      >
        <div
          class="progress absolute top-0 bottom-4 left-0 right-0"
        >
          <div
            class="progress-fill w-full h-full"
            style="transform: scaleY(0);"
          />
        </div>
        <div
          class="absolute left-[50%] translate-x-[-6px] w-[16px] rounded-full bg-[#FF1464] h-[16px] top-0 bottom-4 right-0"
          style="top: 0%;"
        />
        <div
          class="flex h-[400px] opacity-50 ui-virtual-common items-center justify-between ui-virtual-box-1"
        >
          <div
            class="w-[50%]"
          >
            <img
              alt="Description"
              class="max-w-[436px] mb-0"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/icons/feature-layers.svg"
              style="color: transparent;"
              width="436"
            />
          </div>
          <div
            class="w-[50%]"
          >
            <div
              class="max-w-[436px] ml-auto"
            >
              <h1
                class="text-xl md-pt-6 my-5 font-bold"
              >
                Text
              </h1>
              <div
                class="text-base text-grey"
              >
                Text
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex h-[400px] opacity-50 ui-virtual-common items-center justify-between ui-virtual-box-2"
        >
          <div
            class="w-[50%]"
          >
            <img
              alt="Description"
              class="max-w-[436px] mb-0"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/icons/feature-layers.svg"
              style="color: transparent;"
              width="436"
            />
          </div>
          <div
            class="w-[50%]"
          >
            <div
              class="max-w-[436px] ml-auto"
            >
              <h1
                class="text-xl md-pt-6 my-5 font-bold"
              >
                Text
              </h1>
              <div
                class="text-base text-grey"
              >
                Text
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex h-[400px] opacity-50 ui-virtual-common items-center justify-between ui-virtual-box-3"
        >
          <div
            class="w-[50%]"
          >
            <img
              alt="Description"
              class="max-w-[436px] mb-0"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/icons/feature-layers.svg"
              style="color: transparent;"
              width="436"
            />
          </div>
          <div
            class="w-[50%]"
          >
            <div
              class="max-w-[436px] ml-auto"
            >
              <h1
                class="text-xl md-pt-6 my-5 font-bold"
              >
                Text
              </h1>
              <div
                class="text-base text-grey"
              >
                Text
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
